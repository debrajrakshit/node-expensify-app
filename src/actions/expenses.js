import uuid from 'uuid';

// ADD_EXPENSE
// desctructure the first argument
// if doesn't exist destructure an empty object - lecture 93 (section 10)
// define all of the stuffs want to grab -  define as empty string or as per object
export const addExpense = (
    { 
        description = '',
        note = '',
        amount= 0,
        createdAt = 0 
    } = {}
) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
});

// REMOVE_EXPENSE
export const removeExpense = ({ id } = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
});

// EDIT_EXPENSE
export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});

