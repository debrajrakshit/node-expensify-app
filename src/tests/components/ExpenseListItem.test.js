import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render ExpneseListItem with expense', () => {
    const expense = expenses[0];
    const wrapper = shallow(<ExpenseListItem {...expense} />);

    expect(wrapper).toMatchSnapshot();
});