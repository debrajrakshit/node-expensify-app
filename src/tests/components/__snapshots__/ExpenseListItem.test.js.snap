// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpneseListItem with expense 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    196
     - 
    0
  </p>
</div>
`;
